### ISSUE TITLE
E2E tests fail when using a private container registry

### ISSUE THREAD

**Issue by ItalyPaleAle** [dapr/dapr#4375]
## In what area(s)?

/area test-and-release

## Description

Currently tests are failing when the test images are stored on a private container registry (ie. one that doesn't allow public, un-authenticated pulls), even when the K8s cluster itself is authenticated against the container registry. Happening on both Linux and Windows.

This was confirmed with using Azure Container Registry, with K8s authenticated using Azure RBAC.

Example of logs: https://pastebin.com/Mb4yWRcM (was too long to include in this issue)

**Comment by ItalyPaleAle** [comment]
@artursouza  can you please triage and assign this to me?

**Comment by ItalyPaleAle** [comment]
I need to do more testing. I suspect it may be related to how AKS clusters on Azure authenticate with ACR, given that it's using some non-standard way. Will report once I have some findings.

**Comment by ItalyPaleAle** [comment]
I troubleshooted this. The reason for the error wasn't due to the container images being private.
Instead, it was caused by the fact that when not running inside the Dapr Test environment in Azure, MongoDB needs to be started in the cluster (which was not explained in the docs).
I've found a couple of issues with the tests and docs while investigating this issue, and I'll open a PR for those. For now, we can close this.
